client/x/evmstaking/keeper/staking_queue_test.go

```diff
@@ -0,0 +1,120 @@
+package keeper_test
+
+import (
+	"context"
+	"time"
+
+	"cosmossdk.io/math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	stypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+
+	"go.uber.org/mock/gomock"
+)
+
+// setupUnbonding creates unbondings for testing.
+func (s *TestSuite) setupUnbonding(ctx context.Context, delAddr sdk.AccAddress, valAddr sdk.ValAddress, amount string) {
+	require := s.Require()
+	bankKeeper, stakingKeeper := s.BankKeeper, s.StakingKeeper
+
+	bankKeeper.EXPECT().SendCoinsFromModuleToModule(gomock.Any(), stypes.BondedPoolName, stypes.NotBondedPoolName, gomock.Any())
+	_, _, err := stakingKeeper.Undelegate(ctx, delAddr, valAddr, math.LegacyMustNewDecFromStr(amount))
+	require.NoError(err)
+}
+
+// setupMaturedUnbonding creates matured unbondings for testing.
+func (s *TestSuite) setupMatureUnbonding(ctx context.Context, delAddr sdk.AccAddress, valAddr sdk.ValAddress, amt string, duration time.Duration) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	pastHeader := sdkCtx.BlockHeader()
+	pastHeader.Time = pastHeader.Time.Add(-duration).Add(-time.Minute)
+	pastCtx := sdkCtx.WithBlockHeader(pastHeader)
+
+	s.setupUnbonding(pastCtx, delAddr, valAddr, amt)
+}
+
+func (s *TestSuite) TestGetMatureUnbondedDelegations() {
+	require := s.Require()
+	ctx, keeper, stakingKeeper := s.Ctx, s.EVMStakingKeeper, s.StakingKeeper
+
+	// Setup unbonding period
+	params, err := stakingKeeper.GetParams(ctx)
+	require.NoError(err)
+	params.UnbondingTime, err = time.ParseDuration("3600s")
+	require.NoError(err)
+	require.NoError(stakingKeeper.SetParams(ctx, params))
+
+	// Setup 2 validators
+	pubKeys, accAddrs, valAddrs := createAddresses(3)
+	delAddr := accAddrs[0]
+	delPubKey := pubKeys[0]
+	valPubKey1 := pubKeys[1]
+	valAddr1 := valAddrs[1]
+	valPubKey2 := pubKeys[2]
+	valAddr2 := valAddrs[2]
+	s.setupValidatorAndDelegation(ctx, valPubKey1, delPubKey, valAddr1, delAddr)
+	s.setupValidatorAndDelegation(ctx, valPubKey2, delPubKey, valAddr2, delAddr)
+
+	// set staking module's params
+	ubdTime := time.Duration(3600) * time.Second
+	stakingParams, err := stakingKeeper.GetParams(ctx)
+	require.NoError(err)
+	stakingParams.UnbondingTime = ubdTime
+	require.NoError(err)
+	require.NoError(stakingKeeper.SetParams(ctx, stakingParams))
+
+	tcs := []struct {
+		name           string
+		setUnbondings  func(c context.Context)
+		expectedResult []stypes.DVPair
+		expectedError  string
+	}{
+		{
+			name: "pass: no matured unbondings",
+			setUnbondings: func(c context.Context) {
+				s.setupUnbonding(c, delAddr, valAddr1, "100")
+				s.setupUnbonding(c, delAddr, valAddr2, "100")
+			},
+			expectedResult: nil,
+		},
+		{
+			name: "pass: one mautred and one not matured unbonding",
```

Comment:
[nit] mautred -> matured

Reply:
@limengformal 
I have rebased and fixed typo. Thanks